### PR TITLE
Add quantized op support to llama runner

### PR DESCRIPTION
**Pull request by larryliu0820**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3062
* #3055

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D56197863](https://our.internmc.facebook.com/intern/diff/D56197863)

### CODE REVIEW

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56197863).

File: .ci/scripts/test_llama.sh [hunk 1]
```diff
@@ -84,7 +91,7 @@ cmake_build_llama_runner() {
         -DEXECUTORCH_BUILD_CUSTOM="$CUSTOM" \
         -DEXECUTORCH_BUILD_OPTIMIZED=ON \
         -DEXECUTORCH_BUILD_XNNPACK="$XNNPACK" \
-        -DEXECUTORCH_BUILD_OPTIMIZED=ON \
+        -DEXECUTORCH_BUILD_QUANTIZED="$QE" \
```
Comment (kimishpatel) [inline .ci/scripts/test_llama.sh:94]: Can we not always build? haveing so many options for build feels like additional burden for users. Maybe do default opt-in?

**Review by mikekgfb — APPROVED** [review]
Thanks!

File: .ci/scripts/test_llama.sh [hunk 2]
```diff
@@ -126,9 +133,15 @@ fi
 # Export model.
 EXPORTED_MODEL_NAME="${EXPORTED_MODEL_NAME}.pte"
 echo "Exporting ${EXPORTED_MODEL_NAME}"
-EXPORT_ARGS="-c stories110M.pt -p ${PARAMS} -d ${DTYPE} -n ${EXPORTED_MODEL_NAME}"
-if [[ "${MODE}" == "xnnpack+kv+custom" ]]; then
-  EXPORT_ARGS="${EXPORT_ARGS} -kv --use_sdpa_with_kv_cache -X -qmode 8da4w -G 128"
+EXPORT_ARGS="-c stories110M.pt -p ${PARAMS} -d ${DTYPE} -n ${EXPORTED_MODEL_NAME} -kv"
+if [[ "${XNNPACK}" == "ON" ]]; then
+  EXPORT_ARGS="${EXPORT_ARGS} -X -qmode 8da4w -G 128"
```
Comment (kimishpatel) [inline .ci/scripts/test_llama.sh:138]: nit: does += operator work?

File: .ci/scripts/test_quantized_aot_lib.sh [hunk 3]
```diff
@@ -24,7 +24,7 @@ build_cmake_quantized_aot_lib() {
     && retry cmake -DBUCK2=buck2 \
       -DCMAKE_BUILD_TYPE=Release \
       -DCMAKE_PREFIX_PATH="$CMAKE_PREFIX_PATH" \
-      -DEXECUTORCH_BUILD_QUANTIZED=ON \
+      -DEXECUTORCH_BUILD_QUANTIZED_OPS_AOT=ON \
```
Comment (kimishpatel) [inline .ci/scripts/test_quantized_aot_lib.sh:27]: same as before. default ON would be good

File: examples/models/llama2/CMakeLists.txt [hunk 4]
```diff
@@ -91,6 +92,7 @@ add_subdirectory(runner)
 if(EXECUTORCH_USE_TIKTOKEN)
   # find RE2 for tokenizer
   set(ABSL_ENABLE_INSTALL ON)
+  set(ABSL_PROPAGATE_CXX_STD ON)
```
Comment (kimishpatel) [inline examples/models/llama2/CMakeLists.txt:95]: oh we depend on abseil for tiktoken?

Comment (lucylq) [inline examples/models/llama2/CMakeLists.txt:95]: no tests using this path yet right?

File: .ci/scripts/test_llama.sh [hunk 5]
```diff
@@ -126,9 +133,15 @@ fi
...
+fi
+if [[ "${CUSTOM}" == "ON" ]]; then
+  EXPORT_ARGS="${EXPORT_ARGS} --use_sdpa_with_kv_cache"
+fi
+if [[ "${QE}" == "ON" ]]; then
+  EXPORT_ARGS="${EXPORT_ARGS} --embedding-quantize 8,1024"
```
Comment (lucylq) [inline .ci/scripts/test_llama.sh:144]: thanks for adding tests!

**Review by lucylq — APPROVED** [review]

File: examples/models/llama2/CMakeLists.txt (hunk 4 above)
Comment (larryliu0820) [reply examples/models/llama2/CMakeLists.txt:95]: Yeah, tiktoken -> re2 -> abseil

Comment (larryliu0820) [reply examples/models/llama2/CMakeLists.txt:95]: Not yet

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56197863).

**Pull request closed without merging**